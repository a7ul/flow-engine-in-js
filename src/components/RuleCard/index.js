import React from 'react';
import PropTypes from 'prop-types';
import Accordion from 'react-super-accordion';

const RuleCard = (props) => (
  <Accordion headText={props.title}>
    <div>
      <div>
        {props.body}
      </div>
      <div>
        {JSON.stringify(props.status)}
      </div>
      <div>
        {props.trueID}
      </div>
      <div>
        {props.falseID}
      </div>
      <div>
        {props.step}
      </div>
    </div>
  </Accordion>
);
RuleCard.defaultProps = {
  title: null,
  status: null,
  body: null,
  trueID: null,
  falseID: null,
  step: null
};
RuleCard.propTypes = {
  title: PropTypes.string,
  status: PropTypes.oneOf([true, false, null]),
  body: PropTypes.string,
  trueID: PropTypes.string,
  falseID: PropTypes.string,
  step: PropTypes.number
};
export default RuleCard;
