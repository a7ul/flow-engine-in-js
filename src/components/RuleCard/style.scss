.rule-card-body{
    background-color: darkgrey;
    border-radius: 0 0 5px 5px;
    display: flex;
    padding: 10px 15px;
    flex-direction: column;
    font-weight: 500;
  
    .rule-body-title{
        color: white;
        padding: 5px 0;
    }
    .rule-body-function {
        padding: 10px;
        font-weight: bold;
        font-family: 'Courier New', Courier, monospace;
        background-color: rgb(238, 235, 235);
    }
    .rule-body-bottom-half{
        display: flex;
        flex-direction: row;
        align-self: stretch;
        align-items: center;
        justify-content: space-between;

        .rule-next-id-container{
            padding: 10px 0;
        }
        .rule-next-id-title{
            color: white;
            padding: 5px 0;
        }
        .rule-next-id-value{
            padding: 5px;
            background-color: rgb(238, 235, 235);        
        }
    }
}