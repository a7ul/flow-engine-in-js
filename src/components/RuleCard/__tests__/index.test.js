import React from 'react';
import renderer from 'react-test-renderer';
import RuleCard from '../index';

it('renders correctly', () => {
  const tree = renderer
    .create(<RuleCard />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

