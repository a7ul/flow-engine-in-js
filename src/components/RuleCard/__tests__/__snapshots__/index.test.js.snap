// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  onClick={[Function]}
  onToggle={[Function]}
  style={
    Object {
      "boxShadow": "0px 0px 10px 0px lightgrey",
      "cursor": "pointer",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "fontFamily": "helvetica",
      "fontWeight": "bold",
      "padding": "10px",
      "userSelect": "none",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  />
  <div
    style={
      Object {
        "textAlign": "right",
        "transform": "rotateZ(-180deg)",
        "transition": "transform 0.25s",
      }
    }
  >
    âŒƒ
  </div>
</div>
`;
